github.cake

```diff
@@ -0,0 +1,78 @@
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Octokit&version=0.48.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=NuGet.Versioning&version=5.7.0"
+#addin "nuget:https://api.nuget.org/v3/index.json?package=Cake.Yaml&version=3.1.1"
+
+using Octokit;
+using NuGet.Versioning;
+
+static CakeGitHubReleaseInfo _cakeGitHubReleaseInfo;
```

Comment:
Instead of using a global variable, we can use a [typed context](https://cakebuild.net/docs/writing-builds/sharing-build-state#typed-context) to share release information across tasks.

Reply:
Is there a rule of thumb for deciding between putting code in a build task or in a utility/extension to `ICakeContext`? I followed the prior art of [nuget.cake](https://github.com/cake-build/website/blob/79122cbfde83fa81334a7e3576c0acb435137cec/nuget.cake) with adding an extension method to `ICakeContext`, so the in-memory cache is an implementation detail of the extension method itself and not a shared context data of the build (i.e. the method returns everything that the task that called it needs, similar to the nuget.cake extensions.

Reply:
I was talking about the global `_cakeGitHubReleaseInfo` variable. I assume the reason why this is a global variable is because the method is called twice, to avoid having to call out twice to GitHub. Instead of a global variable, the method can be called once and the result be stored in a typed context which then can be passed across the tasks. But I'm OK to keep it like this for the moment. We can look at refactoring the build process in a later step (especially since we didn't use any typed context until now)

Reply:
I'm happy to refactor into a build task with shared context as that's not something I've used before, so a good excuse to try it out.

The `_cakeGitHubReleaseInfo` is not used anywhere but in the extension method so whilst technically a global variable, it's not accessed by any other part of the build - that's what I meant.

Reply:
I still think we can merge this first and do the refactoring later. Also to keep this PR simpler and more focused. Introduction of a typed context would touch a lot of stuff in the build script not related to showing the version on home page. And from a functional point of view the implementation works fine.